pkg/core/http/cors.go

```diff
@@ -21,9 +42,26 @@ func FixupCORSHeaders(downstream http.ResponseWriter, upstream *http.Response) {
 	// Upstream has provided CORS header; upstream will manage all CORS headers
 	// Remove existing CORS headers from response to downstream
 	headers := downstream.Header()
-	for name := range headers {
+	for name, values := range headers {
 		if strings.HasPrefix(name, "Access-Control-") {
 			headers.Del(name)
 		}
+		// Delete 'Vary: Origin' header
+		if name == "Vary" {
```

Comment:
Would like to in this change, are we want to avoid duplicate value in `Vary` header? Why we need to target `Origin`? Or we should get the union of upstream  and downstream vary list, and remove them from the downstream vary list?

Reply:
Since this function targets headers injected by CORS middleware, so its main purpose is to remove duplicated CORS related headers (i.e. `Vary: Origin`). We'll need to revisit header handling in future if gateway would inject other headers in the response.

Reply:
Since gateway will proxy to microservice, assume microservice response headers has `Access-Control-` only but not `Vary`, `hasCORSHeaders` will become true. Then we will remove `Origin` from down stream. Am I understand correctly and is it a problem?

Reply:
`Vary: Origin` indicates some response header may change according to request Origin. In our case, we may emit different `Access-Control-Allowed-Origin` header. If microservices has chosen to manage the CORS header themselves, they are responsible for emitted correct CORS related headers, including `Vary: Origin` if necessary.

Reply:
Would you mind to add some comment here we have this assumption?
LGTM, thanks:)

Reply:
Comment added.